doc/src/tutorial/assumptions.rst

```diff
@@ -0,0 +1,1836 @@
+    >>> x2
+    x
+    >>> x1 == x2
+    True
+
+However if the symbols have different assumptions then they will be considered
+to represent distinct symbols:
+
+    >>> x1 = Symbol('x', positive=True)
+    >>> x2 = Symbol('x', )
+    >>> x1
+    x
+    >>> x2
+    x
+    >>> x1 == x2
+    False
+
+One way to simplify an expression is to use the :func:`~.posify` function
+which will replace all symbols in an expression with symbols that have the
+assumption ``positive=True`` (unless that contradicts any existing assumptions
+for the symbol):
+
+    >>> from sympy import posify, exp
+    >>> x = Symbol('x')
+    >>> expr = exp(sqrt(x**2))
+    >>> expr
+    exp(sqrt(x**2))
+    >>> posify(expr)
+    (exp(_x), {_x: x})
+    >>> expr2, rep = posify(expr)
+    >>> expr2
+    exp(_x)
+
+The :func:`~.posify` function returns the expression with all symbols replaced
+(which might lead to simplifications) and also a dict which maps the new
+symbols to the old that can be used with :py:meth:`~.Basic.subs`. This is
+useful because otherwise the new expression with the new symbols having the
+``positive=True`` assumption will not compare equal to the old:
+
+    >>> expr2
+    exp(_x)
+    >>> expr2 == exp(x)
+    False
+    >>> expr2.subs(rep)
+    exp(x)
+    >>> expr2.subs(rep) == exp(x)
+    True
+
+
+Applying assumptions to string inputs
+=====================================
+
+We have seen how to set assumptions when calling the :class:`~.Symbol` or
+:func:`~.symbols` functions explicitly. A natural question to ask is in what
+other situations can we assign assumptions to an object?
+
+It is common for users to use strings as input to SymPy functions (although
+the general feeling among SymPy developers is that this should be discouraged)
+e.g.:
+
+    >>> from sympy import solve
+    >>> solve('x**2 - 1')
+    [-1, 1]
+
+When creating symbols explicitly it would be possible to assign assumptions
+that would affect the behaviour of :func:`~.solve`:
+
+    >>> x = Symbol('x', positive=True)
+    >>> solve(x**2 - 1)
+    [1]
+
+When using string input SymPy will create the expression and create all of the
+symbolc implicitly so the question arises how can the assumptions be
+specified? The answer is that rather than depending on implicit string
+cponversion it is better to use the :func:`~.parse_expr` function explicitly
+and then it is possible to provide assumptions for the symbols e.g.:
+
+    >>> from sympy import parse_expr
+    >>> parse_expr('x**2 - 1')
+    x**2 - 1
+    >>> eq = parse_expr('x**2 - 1', {'x':Symbol('x', positive=True)})
+    >>> solve(eq)
+    [1]
+
+.. note:: The :func:`~.solve` function is unusual as a high level API in that it
+          actually checks the assumptions on any input symbols (the unknowns)
+          and uses that to tailor its output. The assumptions system otherwise
+          affects low-level evaluation but is not necessarily handled
+          explicitly by high-level APIs.
+
+
+Predicates
+==========
```

Comment:
I would move this section in to the assumptions documentation rather than having it in the tutorial.

Reply:
Same with most if not all the sections below this one. The tutorial should just be basic introductory stuff for new users.